pkg/pub_validations/lib/html/html_validation.dart

```diff
@@ -101,6 +101,18 @@ void _validateHead(Element head) {
   if (canonicalLinks.length > 1) {
     throw ArgumentError('More than one canonical link was specified.');
   }
+  if (canonicalLinks.isEmpty) {
```

Comment:
nit: Maybe we should add a comment explaining: 
 (A) what is a canonical link,
 (B) What is a noindex meta tag,
 (C) Why should one of them be present

I don't disagree that it's reasonable, but it might be nice to explain future readers.